causal_testing/specification/causal_dag.py

```diff
@@ -133,7 +133,9 @@ class CausalDAG(nx.DiGraph):
     def __init__(self, dot_path: str = None, **attr):
         super().__init__(**attr)
         if dot_path:
-            pydot_graph = pydot.graph_from_dot_file(dot_path)
+            with open(dot_path, 'r', encoding='utf-8') as file:
+                dot_content = file.read().replace('\n', '')
```

Comment:
I think it may be worth adding a comment describing this, as the problem this is solving is not obvious from the code.